### ISSUE TITLE
Question about the performance on paper

### ISSUE THREAD

**Issue by lluowoull** [Zhang-VISLab/Learning-to-Segment-3D-Point-Clouds-in-2D-Image-Space#13]
Hi, thanks for providing the code, but I have a question about the performance on paper
![90972589-fc379e00-e54c-11ea-8297-aba691fcd83a](https://user-images.githubusercontent.com/71194181/93048435-dd4c9900-f62c-11ea-8a11-da1ec42b7a12.png)
Why the mIoU does not equal to the average of each individual class? Is there something wrong in code?
The result provided here is a bit low compared with paper, so if the result in paper is not correct, what is the true performance of this model?

**Comment by YechengLyu** [comment]
Thanks to point it out. the IoU of **bag** category should be **88.9**. Sorry for the typo.

Best,
Yecheng

**Comment by lluowoull** [comment]
> Thanks to point it out. the IoU of **bag** category should be **88.9**. Sorry for the typo.
> 
> Best,
> Yecheng

How do you know that? If you can check that, why not providing the checkpoint to others?

I'm not sure should I spend time to follow your model, the mistake in the paper and irreproducible reports from the others make me doubt if your model is **truly** working or not.  Really hope you can provide that

**Comment by Zhang-VISLab** [comment]
Dear lluowoull,

Thanks for your great comments. We are terribly sorry that we lost our CVPR 2020 code after submission. This repository is a reproduced work, and we released a pre-trained network model with 88.0% instance-mean-iou and 86.5% class-mean-iou. An updated ArXiv preprint is available. Could you please check our code to see whether there still exist some issues?

**Comment by Zhang-VISLab** [comment]
Please open a new issue if there is still an issue. Thanks.

**Comment by lluowoull** [comment]
> Please open a new issue if there is still an issue. Thanks.

Thanks for the update, the code is clear and easy to use.
I can't train it from scratch as I'm not working on this dataset, I just want to pick a model for my own task.  I quickly download your prepared data and checkpoint, the test looks good as you report, then I switch to evaluate the training set since that is the easiest way to check further. Then I find the performance on training is very bad. I guess it's caused by data augmentation, so I prepare the training without augmentation with repeat number 1. The test result on training is still not good, I get around 0.75 class iou and 0.81 instance iou. For me, it looks like the provided checkpoint is an overfitted one on the test set. I'm not sure if I did every thing correctly, but I guess you may want to verify that cuz it's really not a good sign.  I didn't open a new issue, sorry about that, as I said, I'm not working on this dataset, so it won't affect me.  If I did something wrong, please ignore me.

**Comment by YechengLyu** [comment]
Thank you for your kind examination and discussion. In this work there are two points may lead to the "overfitting" you mentioned. The first one is, during training we save the weights with best performance on the validation set instead of the training set, which may result in the better performance on testing set. The second one is, in our testing code, we further filter the semantic prediction according to the object class, e.g. any point from a plane object is expected within category 1,2,3,4, according to the ShapeNet dataset.

**Comment by lluowoull** [comment]
> Thank you for your kind examination and discussion. In this work there are two points may lead to the "overfitting" you mentioned. The first one is, during training we save the weights with best performance on the validation set instead of the training set, which may result in the better performance on testing set. The second one is, in our testing code, we further filter the semantic prediction according to the object class, e.g. any point from a plane object is expected within category 1,2,3,4, according to the ShapeNet dataset.

1. I saw you merge the val into train, but even you use test as val, a much worse result on training is still unusual. 

2. I'm using your test code, the only thing I change is the dataset keyword, replace ['test] with ['train'] 

You don't need to argue with me, as I don't want to dispute something or train it from scratch, this is just my feeling after quick check your code.
